CRM/Contribute/BAO/Contribution.php

```diff
@@ -493,91 +497,45 @@ public static function create(&$params, $ids = []) {
     }
 
     $params['contribution_id'] = $contribution->id;
-    $session = CRM_Core_Session::singleton();
 
     if (!empty($params['note'])) {
       $noteParams = [
         'entity_table' => 'civicrm_contribution',
         'note' => $params['note'],
         'entity_id' => $contribution->id,
-        'contact_id' => $session->get('userID'),
+        'contact_id' => CRM_Core_Session::getLoggedInContactID(),
       ];
       if (!$noteParams['contact_id']) {
         $noteParams['contact_id'] = $params['contact_id'];
       }
       CRM_Core_BAO_Note::add($noteParams);
     }
 
-    // make entry in batch entity batch table
```

Comment:
So all of this is to do a load if we don't have all the contribution fields - maybe a helper function that identifies if there IS missing required info